libraries/AP_Scripting/AP_Scripting.cpp

```diff
@@ -197,17 +205,43 @@ void AP_Scripting::repl_stop(void) {
 }
 
 void AP_Scripting::thread(void) {
-    lua_scripts *lua = new lua_scripts(_script_vm_exec_count, _script_heap_size, _debug_level, terminal);
-    if (lua == nullptr || !lua->heap_allocated()) {
-        gcs().send_text(MAV_SEVERITY_CRITICAL, "Unable to allocate scripting memory");
+    while (true) {
+        // reset flags
+        _stop = false;
+        _restart = false;
+
+        lua_scripts *lua = new lua_scripts(_script_vm_exec_count, _script_heap_size, _debug_level, terminal);
+        if (lua == nullptr || !lua->heap_allocated()) {
+            gcs().send_text(MAV_SEVERITY_CRITICAL, "Unable to allocate scripting memory");
+            _init_failed = true;
+        } else {
+            // run won't return while scripting is still active
+            lua->run();
+
+            // only reachable if the lua backend has died for any reason
+            gcs().send_text(MAV_SEVERITY_CRITICAL, "Scripting has stopped");
+        }
         delete lua;
-        _init_failed = true;
-        return;
-    }
-    lua->run();
 
-    // only reachable if the lua backend has died for any reason
-    gcs().send_text(MAV_SEVERITY_CRITICAL, "Scripting has stopped");
+        bool cleared = false;
+        while(true) {
```

Comment:
nitpick; add a little space between "while" and "("

Reply:
that change is not in my PR, it was from Pete